### PR TITLE
Wrong solver settings when switching the target

### PR DESCRIPTION
**Pull request by mlandres**
[bsc#1196368](https://bugzilla.suse.com/show_bug.cgi?id=1196368)

### CODE REVIEW

File: zypp/solver/detail/Resolver.cc [hunk 1]
```diff
@@ -100,15 +112,42 @@ Resolver::~Resolver()
...
 
+
+void Resolver::setDefaultSolverFlags( bool all_r )
+{
+  MIL << "setDefaultSolverFlags all=" << all_r << endl;
+
+  if ( all_r || _applyDefault_focus ) setFocus( ResolverFocus::Default );
+
+#define ZOLV_FLAG_DEFAULT( ISDEFAULT, ZSETTER )        \
+  if ( all_r || ISDEFAULT ) ZSETTER( indeterminate )
```
Comment (bzeller) [inline zypp/solver/detail/Resolver.cc:123]: Using template functions are cleaner than defines. We should consider using those if possible

**Review by bzeller — COMMENTED** [review]
Looks good aside from a few small comments. We also could think about a testcase for this

File: zypp/ZConfig.cc [hunk 2]
```diff
@@ -812,10 +845,12 @@ namespace zypp
   ZConfig::~ZConfig( )
   {}
 
+  void ZConfig::sigTargetChanged()
```
[the remark this answers is not in the document]
Comment (mlandres) [reply zypp/ZConfig.cc:848]: Yes, I'll change it.

File: zypp/solver/detail/Resolver.cc (hunk 1 above)
Comment (mlandres) [reply zypp/solver/detail/Resolver.cc:123]: This way It's in sync with the setter/getter defines. It strengthens the desire that all these bit-Flags should follow the same schema.  I would agree if such a macro was used at various locations throughout the code, but it's a continuous block and IMO leaves litte room for misuse.
```
#define ZOLV_FLAG_DEFAULT( ISDEFAULT, ZSETTER )        \
  if ( all_r || ISDEFAULT ) ZSETTER( indeterminate )

  ZOLV_FLAG_DEFAULT( _applyDefault_forceResolve,         setForceResolve         );
  ...
  ZOLV_FLAG_DEFAULT( _applyDefault_dupAllowVendorChange, dupSetAllowVendorChange );
#undef ZOLV_FLAG_TRIBOOL
```

**Pull request merged**